packages/eas-cli/src/project/publish.ts

```diff
@@ -0,0 +1,308 @@
+import { Platform } from '@expo/config';
+import crypto from 'crypto';
+import fs from 'fs';
+import { uniqBy } from 'lodash';
+import path from 'path';
+
+import { PartialManifestAsset } from '../graphql/generated';
+import { PublishMutation } from '../graphql/mutations/PublishMutation';
+import { PublishQuery } from '../graphql/queries/PublishQuery';
+import { PresignedPost, uploadWithPresignedPostAsync } from '../uploads';
+
+export const TIMEOUT_LIMIT = 60_000; // 1 minute
+let STORAGE_BUCKET: string;
+export const Platforms: PublishPlatforms[] = ['android', 'ios']; // TODO-JJ allow users to specify this in app.js
+
+if (process.env.NODE_ENV === 'test') {
+  STORAGE_BUCKET = 'update-assets-testing';
+} else if (process.env.EXPO_STAGING || process.env.EXPO_LOCAL) {
+  STORAGE_BUCKET = 'update-assets-staging';
+} else {
+  STORAGE_BUCKET = 'update-assets-production';
+}
+
+export type PublishPlatforms = Extract<'android' | 'ios', Platform>;
+export type RawAsset = {
+  type: string;
+  contentType: string;
+  buffer: Buffer;
+};
+type CollectedAssets = {
+  [platform in PublishPlatforms]?: {
+    launchAsset: RawAsset;
+    assets: RawAsset[];
+  };
+};
+
+export function guessContentTypeFromExtension(ext?: string): string {
```

Comment:
The content type is important for asset storage. In the future we may ask the user to specify the contentType if we don't recognize it. For now we default to 'application/octet-stream'